src/chain/mod.rs

```diff
@@ -54,6 +54,7 @@ pub async fn export<D: Digest>(
 
     // Flush to ensure everything has been successfully written
     writer.flush().await.context("failed to flush")?;
+    writer.shutdown().await.context("failed to shutdown")?;
```

Comment:
This code literally does nothing. `shutdown` is only defined on sockets.

Reply:
Reading the code I was thinking this will should call internally `poll_flush` method from `AsyncWrite` trait in `BufWriter`.